cmd/agent/agent.go

```diff
@@ -0,0 +1,118 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+
+	// Adds version information
+
+	"github.com/grafana/agent/pkg/integrations"
+	"github.com/grafana/agent/pkg/loki"
+	"github.com/grafana/agent/pkg/tempo"
+
+	"github.com/go-kit/kit/log/level"
+	"github.com/grafana/agent/pkg/config"
+	"github.com/grafana/agent/pkg/prom"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/weaveworks/common/server"
+
+	"github.com/go-kit/kit/log"
+)
+
+type AgentServer struct {
+	promMetrics *prom.Agent
+	lokiLogs    *loki.Loki
+	tempoTraces *tempo.Tempo
+	manager     *integrations.Manager
+	srv         *server.Server
+}
+
+func NewAgentServer(logger log.Logger, cfg *config.Config) (*AgentServer, error) {
+	var (
+		promMetrics *prom.Agent
+		lokiLogs    *loki.Loki
+		tempoTraces *tempo.Tempo
+		manager     *integrations.Manager
+	)
+
+	srv, err := server.New(cfg.Server)
+	if err != nil {
+		level.Error(logger).Log("msg", "failed to create server", "err", err)
+		return nil, err
```

Comment:
You can remove the logs here (and at the other return points) and just wrap the message instead

Reply:
Do you mean in the function caller? I considered that but would lose the extra information that each error log has.

Reply:
Yeah, returning `return nil, fmt.Errorf("failed to create server: %w", err)` should be plenty. We'll lose the line where the error happened, but it's pretty easy to trace back by searching since it's a fairly unique message.